lib/ui/shared/overflow_menu.dart

```diff
@@ -84,15 +87,65 @@ class _OverflowMenuState extends ConsumerState<OverflowMenu> {
   void _enterFullscreen() =>
       ref.read(WorkspaceState.provider.notifier).toggleFullscreen(true);
 
+  Future<bool> _showOverWriteDialog() async {
+    return await showConfirmationDialog(context) ?? false;
+  }
+
+  Future<bool> _deleteFileAndProject(CatrobatImageMetaData imageData,
+      ProjectDatabase db, IFileService fileService) async {
+    final fileName = '${imageData.name}.${imageData.format.extension}';
+
+    final result = await fileService.deleteFileInApplicationDirectory(fileName);
+    if (result is Err) {
+      Fluttertoast.showToast(msg: 'Overwriting failed!');
+      return false;
+    }
+
+    Fluttertoast.showToast(msg: 'Overwriting file...');
+
+    final oldProject = await db.projectDAO.getProjectByName(imageData.name);
+    if (oldProject != null && oldProject.id != null) {
+      await db.projectDAO.deleteProject(oldProject.id!);
+      ref.invalidate(ProjectDatabase.provider);
+    }
+
+    return true;
+  }
+
+  Future<bool> _handleCatrobatImageSaveAndOverwrite(
+      CatrobatImageMetaData imageData, ProjectDatabase db) async {
+    final fileService = ref.watch(IFileService.provider);
+    final fileName = '${imageData.name}.${imageData.format.extension}';
+    final fileExists =
+        await fileService.checkIfFileExistsInApplicationDirectory(fileName);
+
+    if (fileExists) {
+      final overWriteCanceled = await _showOverWriteDialog();
+      if (overWriteCanceled) {
+        Fluttertoast.showToast(msg: 'Project not saved!');
+        return false;
+      }
+      return await _deleteFileAndProject(imageData, db, fileService);
+    }
+
+    return true;
+  }
+
   Future<void> _saveProject() async {
-    File? savedProject;
-    final imageData = await showSaveImageDialog(context, true);
+    final imageData =
+        await showSaveImageDialog(context, true) as CatrobatImageMetaData;
```

Comment:
please check what happens if the dialog returns null

Reply:
Added a null check.